monero/utils.go

```diff
@@ -10,44 +11,47 @@ import (
 	logging "github.com/ipfs/go-log"
 )
 
-const (
-	maxRetries         = 360
+var (
+	// blockSleepDuration is the duration that we sleep between checks for new blocks. We
+	// lower it in dev environments if fast background mining is started.
 	blockSleepDuration = time.Second * 10
-)
 
-var (
 	log = logging.Logger("monero")
 )
 
 // WaitForBlocks waits for `count` new blocks to arrive.
 // It returns the height of the chain.
-func WaitForBlocks(client WalletClient, count int) (uint64, error) {
-	prevHeight, err := client.GetHeight()
+func WaitForBlocks(ctx context.Context, client WalletClient, count int) (uint64, error) {
+	startHeight, err := client.GetHeight()
 	if err != nil {
 		return 0, fmt.Errorf("failed to get height: %w", err)
 	}
+	prevHeight := startHeight - 1 // prevHeight is only for logging
+	endHeight := startHeight + uint64(count)
 
-	for j := 0; j < count; j++ {
-		for i := 0; i < maxRetries; i++ {
-			if err := client.Refresh(); err != nil {
-				return 0, err
-			}
+	for {
+		if err := client.Refresh(); err != nil {
+			return 0, err
+		}
 
-			height, err := client.GetHeight()
-			if err != nil {
-				continue
-			}
+		height, err := client.GetHeight()
+		if err != nil {
+			return 0, err
```

Comment:
It didn't make sense for `Refresh` to terminate the function on error and then have `GetHeight` behave differently, so I just simplified and converted 2 loops into 1.